### PR TITLE
[master] Upgrade Quarkus to 1.12.2.Final and some workarounds removed

### PR DESCRIPTION
**Pull request by Sgitario**
- Upgrade Quarkus to 1.12.2.Final
- Upgrade quarkus-qpid-jms to 0.23.0
- Removed workaround for https://github.com/quarkusio/quarkus/issues/14525
- Added commit to update the TODO apps. Cherry-pick for https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/222

### CODE REVIEW

**Comment by Sgitario** [comment]
In Draft, until we verify 1.11.0.ER1.

**Comment by rsvoboda** [comment]
@Sgitario can we proceed with this PR? There is 1.11 branch for RHBQ related testing

**Comment by Sgitario** [comment]
> @Sgitario can we proceed with this PR? There is 1.11 branch for RHBQ related testing

I've rebased the branch. Let's wait for the Jenkins jobs and if they work, let's merge this.

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
Github CI in Native is failing because of the Mandrel issue: https://github.com/graalvm/mandrel/issues/218
Waiting for the rest.

**Comment by Sgitario** [comment]
The target JAR file changed and some more adjustments were needed.
For Native, the mandrel 20.3 image was not allowed anylonger, using the ubi image temporary.

**Comment by Sgitario** [comment]
There were other failures... investigating.

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
After upgrading to 1.12.1.Final, new failures:
- messaging/amqp-reactive
- infinispan-client
- http/http-advanced (only vertxHttpClientRedirection and httpServer)
- deployment-strategies/quarkus
- deployment-strategies/quarkus-serverless: *expected as knative is not working in OCP 4.6 at this moment*

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
There are still two test cases failing in `http-advanced`. I'm trying to investige it. cc @pjgg

**Comment by Sgitario** [comment]
http-advanced: raised https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/224 for fixing

**Comment by Sgitario** [comment]
run tests

File: pom.xml [hunk 1]
```diff
@@ -79,7 +79,7 @@
         <version.maven-compiler-plugin>3.8.1</version.maven-compiler-plugin>
         <version.maven-jar-plugin>3.2.0</version.maven-jar-plugin>
         <version.maven-surefire-plugin>2.22.2</version.maven-surefire-plugin>
-        <version.quarkus>1.11.1.Final</version.quarkus>
+        <version.quarkus>1.12.0.Final</version.quarkus>
```
Comment (rsvoboda) [inline pom.xml:82]: Please use `1.12.1`

File: pom.xml [hunk 2]
```diff
@@ -322,7 +322,9 @@
 
             <properties>
                 <quarkus.package.type>native</quarkus.package.type>
-                <quarkus.native.builder-image>quay.io/quarkus/ubi-quarkus-mandrel:20.1.0.1.Final-java11</quarkus.native.builder-image>
+                <!-- TODO: Can't use `ubi-quarkus-mandrel` image due to https://github.com/graalvm/mandrel/issues/218 -->
+                <!-- <quarkus.native.builder-image>quay.io/quarkus/ubi-quarkus-mandrel:20.1.0.1.Final-java11</quarkus.native.builder-image> -->
```
Comment (rsvoboda) [inline pom.xml:326]: and the image issue is probably resolved, like one day ago or so :)

Comment (Sgitario) [reply pom.xml:326]: @mjurc said to use the mandrel version from the pom.xml. Either I should remove this comment or use the mandrel version? What do you think?

File: pom.xml (hunk 1 above)
Comment (Sgitario) [reply pom.xml:82]: We can't use 1.12.1 version because of the issue with OpenShift.

File: pom.xml (hunk 2 above)
Comment (rsvoboda) [reply pom.xml:326]: I was reacting to Mandrel issue which is resolved now as new images were published 18 hours ago.

"mandrel version from pom" ... you probably mean the default builder image defined in upstream Quarkus

I think we should continue with Mandrel here as it is the default for product.

@mjurc introduced Mandrel image in his PR https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/63, would be interested to hear what changed his mind wrt the default builder image :)

Comment (Sgitario) [reply pom.xml]: I've changed this to use the mandrel 21.0 image version. Let me know if you want to use the version from the pom.xml instead.

**Comment by rsvoboda** [comment]
1.12.2.Final seems to be on the way

https://repo1.maven.org/maven2/io/quarkus/quarkus-bom/1.12.2.Final/ is available
https://repo1.maven.org/maven2/io/quarkus/quarkus-universe-bom/1.12.2.Final/ is not yet available

**Comment by rsvoboda** [comment]
https://repo1.maven.org/maven2/io/quarkus/quarkus-universe-bom/1.12.2.Final/ is available

Please update the PR to use 1.12.2.Final.

**Comment by Sgitario** [comment]
> https://repo1.maven.org/maven2/io/quarkus/quarkus-universe-bom/1.12.2.Final/ is available
> 
> Please update the PR to use 1.12.2.Final.

Updated

**Comment by rsvoboda** [comment]
GH action runs look good, let's wait for OCP run

**Comment by Sgitario** [comment]
> GH action runs look good, let's wait for OCP run

Too many failures due to some network issues. Trying again:

run tests

**Comment by rsvoboda** [comment]
run tests

**Comment by rsvoboda** [comment]
Update from Jose:

I know about the 1.12.2 issues:
```
[0;33mWARN ARTIFACT_COPY_ARGS is deprecated.  Please use S2I_SOURCE_DEPLOYMENTS_FILTER to specify artifact types and MAVEN_S2I_ARTIFACT_DIRS to specify the build output directories to copy from.[0m
cp: cannot stat 'lib/': No such file or directory
'rest-villain-01-runner.jar' -> '/deployments/rest-villain-01-runner.jar'
error: build error: error building at STEP "RUN /usr/local/s2i/assemble": error while running runtime: exit status 1
```

Jose Carvajal Hilario, Now, Edited
but I need to focus on OCP 3.11 first

**Comment by Sgitario** [comment]
> Update from Jose:
> 
> I know about the 1.12.2 issues:
> 
> ```
> �[0;33mWARN ARTIFACT_COPY_ARGS is deprecated.  Please use S2I_SOURCE_DEPLOYMENTS_FILTER to specify artifact types and MAVEN_S2I_ARTIFACT_DIRS to specify the build output directories to copy from.�[0m
> cp: cannot stat 'lib/': No such file or directory
> 'rest-villain-01-runner.jar' -> '/deployments/rest-villain-01-runner.jar'
> error: build error: error building at STEP "RUN /usr/local/s2i/assemble": error while running runtime: exit status 1
> ```
> 
> Jose Carvajal Hilario, Now, Edited
> but I need to focus on OCP 3.11 first

It seems that the folder `/lib` is not needed any longer. After removing it, it works fine. 
I've also rebased the branch.

File: config-secret/file-system/src/main/resources/application.properties [hunk 3]
```diff
@@ -3,7 +3,4 @@ quarkus.openshift.app-secret=app-config
 
 quarkus.s2i.base-jvm-image=registry.access.redhat.com/ubi8/openjdk-11:latest
 
-%test.hello.message=Hello, %s!
-
-# the application binary resides in /home/quarkus with quarkus/ubi-quarkus-native-binary-s2i
-%native.quarkus.openshift.mounts.app-config.path=/home/quarkus/config
```
Comment (rsvoboda) [inline config-secret/file-system/src/main/resources/application.properties:12]: We should probably track these details somewhere upstream or at least document how these paths can be figured out

**Comment by rsvoboda** [comment]
```
Error:  Failed to execute goal on project sql-db-app: Could not resolve dependencies for project io.quarkus.ts.openshift:sql-db-app:jar:1.0.0-SNAPSHOT: Could not transfer artifact io.quarkus:quarkus-agroal:jar:1.12.2.Final from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/io/quarkus/quarkus-agroal/1.12.2.Final/quarkus-agroal-1.12.2.Final.jar: Connection reset -> [Help 1]
```
Time for custom setting with google mirror instead of apache central?

File: config-secret/file-system/src/main/resources/application.properties (hunk 3 above)
Comment (Sgitario) [reply config-secret/file-system/src/main/resources/application.properties:12]: I'm not sure what you mean. I checked the git history and this was added as a workaround for an upstream issue that is now fixed. That's why I removed it.

**Comment by Sgitario** [comment]
> ```
> Error:  Failed to execute goal on project sql-db-app: Could not resolve dependencies for project 
> io.quarkus.ts.openshift:sql-db-app:jar:1.0.0-SNAPSHOT: Could not transfer artifact io.quarkus:quarkus-
> agroal:jar:1.12.2.Final from/to central (https://repo.maven.apache.org/maven2): Transfer failed for 
> https://repo.maven.apache.org/maven2/io/quarkus/quarkus-agroal/1.12.2.Final/quarkus-agroal-1.12.2.Final.jar:
>  Connection reset -> [Help 1]
> ```
> 
> Time for custom setting with google mirror instead of apache central?

Do you want me to do this in the OpenShift TS jenkins jobs?

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
@rsvoboda I think this PR is ready to be merged. The native job didn't finish because some issues with the Jenkins node:

```
00:50:11 ERROR: Build step failed with exception
00:50:11 java.lang.NullPointerException: no workspace from node jenkins.plugins.openstack.compute.JCloudsSlave[rhel8-large-60] which is computer jenkins.plugins.openstack.compute.JCloudsComputer@582572d9 and has channel null
```

what do you think? 

If merged, please, don't squash the commits.

**Review by rsvoboda — APPROVED** [review]
Approving, thanks for handling all the resolved issues and identifying regressions which popped in 1.12.0.Final and 1.12.1.Final. I think we can merge this PR. I was a long ride!

**Pull request merged**